controllers/story.js

```diff
@@ -1,431 +1,449 @@
-        }
+    story = story.pop();
+    var dashedNameFull = story.storyLink.toLowerCase().replace(/\s/g, '-');
+    if (dashedNameFull !== dashedName) {
+      return res.redirect('../stories/' + dashedNameFull);
+    }
 
-		var userVoted = false;
-        try {
-            var votedObj = story.upVotes.filter(function(a){
-                return a['upVotedByUsername'] === req.user['profile']['username'];
-            })
-            if (votedObj.length > 0){
-                userVoted = true;
-            }
-        } catch(err){
-            userVoted = false;
-        }
-	res.render('stories/index', {
-            title: story.headline,
-            link: story.link,
-            author: story.author,
-            description: story.description,
-            rank: story.upVotes.length,
-            upVotes: story.upVotes,
-            comments: story.comments,
-            id: story._id,
-            timeAgo: moment(story.timePosted).fromNow(),
-            image: story.image,
-            page: 'show',
-            storyMetaDescription: story.metaDescription,
-			hasUserVoted: userVoted
-        });
+    var userVoted = false;
+    try {
+      var votedObj = story.upVotes.filter(function(a) {
+        return a['upVotedByUsername'] === req.user['profile']['username'];
+      });
+      if (votedObj.length > 0) {
+        userVoted = true;
+      }
+    } catch(err) {
+      userVoted = false;
+    }
+    res.render('stories/index', {
+      title: story.headline,
+      link: story.link,
+      author: story.author,
+      description: story.description,
+      rank: story.upVotes.length,
+      upVotes: story.upVotes,
+      comments: story.comments,
+      id: story._id,
+      timeAgo: moment(story.timePosted).fromNow(),
+      image: story.image,
+      page: 'show',
+      storyMetaDescription: story.metaDescription,
+      hasUserVoted: userVoted
     });
+  });
 };
 
 exports.getStories = function(req, res) {
-    MongoClient.connect(secrets.db, function(err, database) {
-        database.collection('stories').find({
-            "$text": {
-                "$search": req.body.data.searchValue
-            }
-        }, {
-            headline: 1,
-            timePosted: 1,
-            link: 1,
-            description: 1,
-            rank: 1,
-            upVotes: 1,
-            author: 1,
-            comments: 1,
-            image: 1,
-            storyLink: 1,
-            metaDescription: 1,
-            textScore: {
-                $meta: "textScore"
-            }
-        }, {
-            sort: {
-                textScore: {
-                    $meta: "textScore"
-                }
-            }
-        }).toArray(function(err, items) {
-            if (items !== null && items.length !== 0) {
-                return res.json(items);
-            }
-            return res.status(404);
-        });
+  MongoClient.connect(secrets.db, function(err, database) {
+    if (err) {
+      return res.status(500);
```

Comment:
need a call to `res.end` as well or the request will never be ended. You could chain this as well `res.status(500).end();`